src/webpack-core/ContainerPlugin.js

```diff
@@ -0,0 +1,186 @@
+const Dependency = require("webpack/lib/Dependency");
+const Module = require("webpack/lib/Module");
+const ModuleDependency = require("webpack/lib/dependencies/ModuleDependency");
+const AsyncDependenciesBlock = require("webpack/lib/AsyncDependenciesBlock");
+const RuntimeGlobals = require("webpack/lib/RuntimeGlobals");
+const ModuleFactory = require("webpack/lib/ModuleFactory");
+const { ConcatSource } = require("webpack-sources");
+
+const PLUGIN_NAME = "ContainerPlugin";
+
+class ContainerExposedDependency extends ModuleDependency {
+  constructor(name, request) {
+    super(request);
+    this._name = name;
+  }
+
+  get exposedName() {
+    return this._name;
+  }
+
+  getResourceIdentifier() {
+    return `exposed dependency ${this._name}`;
+  }
+}
+
+class ContainerEntryDependency extends Dependency {
+  constructor(dependencies, name) {
+    super();
+    this.exposedDependencies = dependencies;
+    this.loc = { name };
+  }
+}
+
+const CONTAINER_ENTRY_MODULE_TYPES = new Set(["javascript"]);
+
+class ContainerEntryModule extends Module {
+  constructor(dependency) {
+    super("javascript/dynamic", null);
+    this.expose = dependency.exposedDependencies;
+  }
+
+  getSourceTypes() {
+    return CONTAINER_ENTRY_MODULE_TYPES;
+  }
+
+  identifier() {
+    return `container entry ${JSON.stringify(
+      this.expose.map(item => item.getResourceIdentifier())
+    )}`;
+  }
+
+  readableIdentifier() {
+    return `container entry`;
+  }
+
+  needBuild(context, callback) {
+    return callback(null, !this.buildMeta);
+  }
+
+  build(options, compilation, resolver, fs, callback) {
+    this.buildMeta = {};
+    this.buildInfo = {
+      strict: true,
+      builtTime: Date.now(),
```

Comment:
You don't need that

Other modules use a build time to determine if the module need to be rebuild. This module never need to be rebuild.

Reply:
Ah perfect! Thought that had something to do with stats or something. Removing! :100: